### ISSUE TITLE
HTTP Publish API Message Issue

### ISSUE THREAD

**Issue by satyendrakumarsingh** [emqx/emqx#9518]
### What happened?

While using HTTP API to publish messages after certain time message not published to connected devices.
Even if I add client into log trace there is no publish message log.

If I use any MQTT client to publish message clients are receiving message properly.

### What did you expect to happen?

If I receive message ID in publish HTTP API call response then message must published on broker so that delivery is completed of published message.

### How can we reproduce it (as minimally and precisely as possible)?

After 12-18 hours this issue occurs and I monitor it for 3 days.

### EMQX version

<details>
```console
$ ./bin/emqx_ctl broker
```
root@notifypro:/home/paresh# /bin/emqx_ctl broker
sysdescr  : EMQX
version   : 5.0.11
datetime  : 2022-12-09T15:36:33.695983830+05:30
uptime    : 20 hours, 24 minutes, 6 seconds

</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
root@notifypro:/home/paresh# cat /etc/os-release
NAME="Ubuntu"
VERSION="20.04.5 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.5 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal

$ uname -a

Linux notifypro 5.4.0-128-generic #144-Ubuntu SMP Tue Sep 20 11:00:04 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

</details>


### Log files

<details>

</details>

**Comment by id** [comment]
@satyendrakumarsingh do you set QoS in the publish request?

**Comment by satyendrakumarsingh** [comment]
Yes I do set QOS= 1 in each HTTP call and this issue occur in running instance after 10-12 hours, I observed this behavior, after re-starting instance also message not published to subscribed devices.

If I use any MQTT desktop client I am able to publish message successfully, when I was using 5.08 for a month I never found this issue, when I upgraded to 5.0.11 I found this issue and later I have also do a fresh installation of 5.0.11 but on next day again I face same issue for all the published messages.

In each publish HTTP call I got message ID in HTTP response but when I enable log trace for CLIENT ID unable to see any publish packets and message also not delivered to devices.
